src/main/java/org/spongepowered/api/event/message/MessageEvent.java

```diff
@@ -22,37 +22,39 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.event.action;
+package org.spongepowered.api.event.message;
 
-import org.spongepowered.api.command.CommandSource;
 import org.spongepowered.api.event.Event;
 import org.spongepowered.api.text.Text;
 
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
 /**
- * Describes events when a {@link CommandSource} sends a {@link Text} message.
+ * Describes events when a involving a {@link Text} message.
  */
 public interface MessageEvent extends Event {
 
     /**
-     * Gets the {@link Text} message created by the {@link CommandSource} before
-     * the calling of this event.
+     * Gets the original {@link Text} message.
      *
-     * @return The message
+     * @return The message if present, otherwise {@link Optional#empty()}
      */
-    Text getOriginalMessage();
+    Optional<Text> getOriginalMessage();
```

Comment:
If a non-player and non-tamed `Entity` is killed, then the message event is triggered but it does not contain a message, but you can set a message so each `Entity` that is killed shows a message like it would have been a `Player`.